### PR TITLE
refactor(card_issuer): List all card issuers

### PR DESCRIPTION
**Pull request by ayush22667**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

`GET /card_issuers` running an `ILIKE` prefix scan over `card_issuers`
with a `limit` (default 30). we’ll return the entire list, since the frontend needs the full list to show all available options and also to map which ones are already blocked.

- `list_card_issuers` reads through `cache::get_or_populate_in_memory` against `CONFIG_CACHE` under
  the key `card_issuers_list`.
- `insert_card_issuer`, `update_card_issuer` and `delete_card_issuer` wrap their DB call in
  `cache::publish_and_redact` with `CacheKind::Config("card_issuers_list")`, so the entry is redacted
  across instances on every write.

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

<details>
<summary>1. get card issuer</summary>

Request
```
curl --location 'http://localhost:8080/card_issuers' \
--header 'Authorization: <REDACTED>' \
--header 'api-key: <REDACTED>'
```

Response
```
{
    "issuers": [
        {
            "id": "HqMed",
            "issuer_name": "(DEFUNCT)"
        },
        {
            "id": "2xPOV",
            "issuer_name": "019 PAYMENT SERVICES LIMITED"
        },
        {
            "id": "Tpne9",
            "issuer_name": "02MONEYCARDS"
        },
        {
            "id": "FxNk1",
            "issuer_name": "100% BANCO, BANCO UNIVERSAL, C.A."
        },
        {
            "id": "yf6Yt",
            "issuer_name": "1199 SEIU FEDERAL CREDIT UNION"
        },
        {
            "id": "JqnGT",
            "issuer_name": "121 FINANCIAL CREDIT UNION"
        },
        {
            "id": "EDp6p",
            "issuer_name": "189286 CANADA INC."
        },
        {
            "id": "swoDC",
            "issuer_name": "1BANCOITAU.S.A."
        },
        {
            "id": "wwXys",
            "issuer_name": "1LINK PRIVATE LIMITED"
        },
        {
            "id": "cwmZw",
            "issuer_name": "1ST ADVANTAGE BANK"
        },
        {
            "id": "RLvVX",
            "issuer_name": "1ST ADVANTAGE FEDERAL CREDIT UNION"
        },
        {
            "id": "Are70",
            "issuer_name": "1ST BANK IN HOMINY"
        },
        {
            "id": "Bd7lb",
            "issuer_name": "1ST BANK OF SEA ISLE CITY"
        },
        {
            "id": "3VscI",
            "issuer_name": "1ST BANK YUMA"
        },
        {
            "id": "VBtKo",
            "issuer_name": "1ST BERGEN FEDERAL CREDIT UNION"
        },
        {
            "id": "zhU2D",
            "issuer_name": "1ST CHOICE BANK"
        },
        {
            "id": "e01S1",
            "issuer_name": "1ST CHOICE COMMUNITY FEDERAL CREDIT UNION"
        },
        {
            "id": "c5N8S",
            "issuer_name": "1ST CHOICE CREDIT UNION"
        },
        {
            "id": "7EgpG",
            "issuer_name": "1ST COLONIAL COMMUNITY BANK"
        },
        {
            "id": "wreUa",
            "issuer_name": "1ST COMMUNITY BANK"
        },
        {
            "id": "CEL7F",
            "issuer_name": "1ST COMMUNITY CREDIT UNION"
        },
        {
            "id": "ScGe9",
            "issuer_name": "1ST COMMUNITY FEDERAL CREDIT UNION"
        },
        {
            "id": "kr4u8",
            "issuer_name": "1ST CONSTITUTION BANK"
        },
        {
            "id": "aAIXU",
            "issuer_name": "1ST COOPERATIVE FEDERAL CREDIT UNION"
        },
        {
            "id": "2tXyv",
            "issuer_name": "1ST ED CREDIT UNION"
        },
        {
            "id": "60veN",
            "issuer_name": "1ST ENTERPRISE BANK"
        },
        {
            "id": "lh4Zz",
            "issuer_name": "1ST EQUITY BANK"
        },
        {
            "id": "9Yoz9",
            "issuer_name": "1ST FEDERAL SAVINGS BANK OF SC, INC."
        },
        {
            "id": "34QHJ",
            "issuer_name": "1ST FINANCIAL BANK USA"
        },
        {
            "id": "N4UrA",
            "issuer_name": "1ST FINANCIAL FEDERAL CREDIT UNION"
        },
        {
            "id": "oSFUz",
            "issuer_name": "1ST GATEWAY CREDIT UNION"
        },
        {
            "id": "JsH4h",
            "issuer_name": "1ST INDEPENDENCE BANK"
        },
        {
            "id": "LtFIC",
            "issuer_name": "1ST KANSAS CREDIT UNION"
        },
        {
            "id": "ETyqp",
            "issuer_name": "1ST LIBERTY FEDERAL CREDIT UNION"
        },
        {
            "id": "4W32I",
            "issuer_name": "1ST MIDAMERICA CREDIT UNION"
        },
        {
            "id": "iZB2q",
            "issuer_name": "1ST MISSISSIPPI FEDERAL CREDIT UNION"
        },
        {
            "id": "EkPdw",
            "issuer_name": "1ST NATIONAL BANK ST. LUCIA LIMITED"
        },
        {
            "id": "3LBsG",
            "issuer_name": "1ST NATIONAL BANK ST. LUCIA LIMITED (ST. VINCENT & GRENADINES)"
        },
        {
            "id": "KDorO",
   
â€¦ [truncated 1828319 more characters]
```
</details>

image

## Linked Issues

- Parent: https://github.com/juspay/hyperswitch-cloud/issues/15049
- Closes https://github.com/juspay/hyperswitch-cloud/issues/23663

## How did you test it?

Cypress: `cypress-tests/cypress/e2e/spec/Misc/00002-CardIssuerManagement.cy.js` covers create, list,
update and delete against a live server. The `listCardIssuers` command was updated to call
`GET /card_issuers` with no query string and to assert that `response.body.issuers` is an array; the
two specs that exercised the removed `query` and `limit` parameters were dropped.

```
npm run cypress:misc
```

Cache behaviour to verify manually: list once (populates the cache), create or update an issuer,
then list again — the new or renamed issuer should appear immediately, confirming the
`publish_and_redact` invalidation on the write paths.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/diesel_models/src/query/card_issuer.rs [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
-        )
-        .await
+    ) -> StorageResult<Vec<CardIssuerListItem>> {
+        dsl::card_issuers
+            .select((dsl::id, dsl::issuer_name))
+            .order(dsl::issuer_name.asc())
+            .get_results_async::<CardIssuerListItem>(conn.raw_connection())
+            .await
+            .change_context(errors::DatabaseError::Others)
+            .attach_printable("Error while listing card issuers")
```
Comment (hrithikesh026) [inline crates/diesel_models/src/query/card_issuer.rs]: should not do list without limit.

Comment (ayush22667) [reply crates/diesel_models/src/query/card_issuer.rs]: Here, the assumption is that we’ll return the entire list, since the frontend needs the full list to show all available options and also to map which ones are already blocked.

**Comment by XyneSpaces** [comment]
[should-fix] `crates/diesel_models/src/query/card_issuer.rs` replaces `list_filtered` with `list_all`, removing the `limit` and `query` parameters and returning the entire `card_issuers` table unbounded. Add a hard upper bound or keep pagination so an unexpectedly large table cannot OOM the response or the in-memory config cache (`CARD_ISSUERS_LIST_CACHE_KEY`).

File: crates/diesel_models/src/query/card_issuer.rs (hunk 1 above)
Comment (jarnura) [reply crates/diesel_models/src/query/card_issuer.rs]: @su-shivanshmathur How this PR is compiling? this should have been a compiler level block right?

Comment (jarnura) [reply crates/diesel_models/src/query/card_issuer.rs]: @ayush22667 We should have a limit but that should have a higher limit.

**Review by jarnura — CHANGES_REQUESTED** [review]

**Comment by XyneSpaces** [comment]
[should-fix] Please avoid silently truncating the “list all card issuers” response at `CARD_ISSUER_LIST_MAX_LIMIT` in `crates/diesel_models/src/query/card_issuer.rs` (added line 19), or add pagination/explicit overflow handling. Once the table exceeds 25,000 rows, this endpoint returns an incomplete list without signaling that entries were omitted.

File: cypress-tests/cypress/e2e/spec/Misc/00002-CardIssuerManagement.cy.js [hunk 2]
```diff
@@ -66,14 +61,6 @@ describe("Card Issuer Management", () => {
   });
 
   context("Edge Case Tests", () => {
-    it("should handle listing with limit of 1", () => {
-      cy.listCardIssuers("", 1, globalState);
-    });
-
-    it("should handle listing with empty query", () => {
```
Comment (hrithikesh026) [inline cypress-tests/cypress/e2e/spec/Misc/00002-CardIssuerManagement.cy.js:73]: this we can keep

**Review by hrithikesh026 — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by likhinbopanna — APPROVED** [review]
